packages/compass-indexes/test/setup-store.ts

```diff
@@ -6,7 +6,36 @@ import type {
 } from '../src/stores/store';
 import configureStore from '../src/stores/store';
 
-export const setupStore = (options: Partial<ConfigureStoreOptions> = {}) => {
+const NOOP_DATA_PROVIDER: IndexesDataService = {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  indexes(ns: string, options: any) {
+    return Promise.resolve([]);
+  },
+  isConnected() {
+    return true;
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  updateCollection(ns: string, flags: any) {
+    return Promise.resolve({});
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  createIndex(ns, spec, options) {
+    return Promise.resolve('ok');
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  dropIndex(ns, name) {
+    return Promise.resolve({});
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  createSearchIndex(ns, name, definition) {
+    return Promise.resolve('');
+  },
+};
+
+export const setupStore = (
+  options: Partial<ConfigureStoreOptions> = {},
+  dataProvider: Partial<IndexesDataService> = NOOP_DATA_PROVIDER
```

Comment:
`dataProvider` is part of options

Reply:
I did it this way because Partial<ConfigureStoreOptions> does not imply that you have Partial<IndexesDataService> inside the ConfigureStoreOptions, and I would like to have some defaults in the data provider that can be overriden per test easily.

Do you see a better way?

Reply:
With deep partial for options and that will allow you to pass methods for data provider that you want to without doing `as any` or simply you can use `as any` when adding mocked data provider with few mocked methods?

Reply:
This method is called by compass-collection, by changing the interface like that, you will break this interface. I also don't think we should have defaults in `createStore` here, this doesn't seem meaningful from the application perspective, why would we want that? If you are doing this only for tests, this logic should stay in tests and not affect application

Reply:
The file is a setupStore only for tests, right? It's in the test folder. Or I am mistaken?

Reply:
Totally misread that file name, I thought it's the main plugin setup store file, you're right 😄